### PR TITLE
[5.2] Fixed isset bug for unloaded relationships

### PR DESCRIPTION
**Pull request by daedeloth**
isset does not work on relationships that have no been loaded
This commit fixes that

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -3460,8 +3460,11 @@ public function offsetUnset($offset)
      */
     public function __isset($key)
     {
-        return (isset($this->attributes[$key]) || isset($this->relations[$key])) ||
-                ($this->hasGetMutator($key) && ! is_null($this->getAttributeValue($key)));
+        return (
+            isset($this->attributes[$key]) ||
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: please keep this part on the same line as it was

Comment (daedeloth) [inline src/Illuminate/Database/Eloquent/Model.php]: This okay?

File: src/Illuminate/Database/Eloquent/Model.php [hunk 2]
```diff
@@ -3460,8 +3460,8 @@ public function offsetUnset($offset)
      */
     public function __isset($key)
     {
-        return (isset($this->attributes[$key]) || isset($this->relations[$key])) ||
-                ($this->hasGetMutator($key) && ! is_null($this->getAttributeValue($key)));
+        return (isset($this->attributes[$key]) || isset($this->relations[$key]) || method_exists($this, $key)) ||
+            ($this->hasGetMutator($key) && ! is_null($this->getAttributeValue($key)));
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: Please don't change the indentation on this line. ;)

**Comment by GrahamCampbell** [comment]
Please add a test case to prove the bug. Integration would be preferred, but both unit and integration if you have time.

**Comment by daedeloth** [comment]
Will do

<https://www.catlab.eu/email/www.catlab.be>
*Thijs Van der Schaeghe*
Alpha Cat
+32 473 72 68 79 <+32473726879> - thijs@catlab.be
Catlab Interactive - Pannestraat 75 - 9000 Gent, België - www.catlab.be -
facebook <https://www.facebook.com/catlab.interactive?fref=ts> - twitter
<https://twitter.com/CatLabGames>

**Comment by daedeloth** [comment]
@GrahamCampbell could you let me know what else is needed?

**Comment by srmklive** [comment]
@daedeloth good work mate. I think only thing @GrahamCampbell might ask you again is to rebase your commits into a single commit.

**Comment by GrahamCampbell** [comment]
> might ask you again is to rebase your commits into a single commit.

Nah, Taylor can just squash on merge.

**Comment by GrahamCampbell** [comment]
Please don't rename the test file.

**Comment by GrahamCampbell** [comment]
It now missmatches the classname.

**Comment by daedeloth** [comment]
Woops not sure how that happened; fixed

**Comment by taylorotwell** [comment]
I question whether this is a bug. Regardless, it could be breaking behavior if people are using isset() to check if relationships have been loaded, and not whether they merely exist. Open to other thoughts but going to close for now since at the least it could break existing applications.

**Comment by daedeloth** [comment]
Well that was an anti climax...

Would people actually be interested to check if relationships have been loaded? 
Without this isset I don't have any way to check if the relationship exists.

If you don't want to merge it in 5.2, at least merge it in master...

**Comment by daedeloth** [comment]
@taylorotwell can we get a bit of a conversation going here? Problem I'm currently having is that I cannot see any proper way around this.

**Pull request closed without merging**